### DISCUSSION TITLE
Demo 2: Use Cases

### DISCUSSION THREAD

**Discussion by JohanCSmit** [COS301-SE-2022/Exploring-Self-Sovereign-Identity#78] [General]
We need to formalize our use cases we are going to show in demo 2 and draw up the diagrams.

**Comment by JohanCSmit** [comment]
@RebeccaPillay Would you be able to assist me with this please?

**Reply by RebeccaPillay** [reply]
I fleshed out the use cases for Demo 2 from my understanding of the current architecture. Let me know if I'm on the right track or if there are major revisions to be made. Once you're happy, I can start on the diagrams. 

1. User Sign-up: User signs up on SSI mobile app (with Biometrics - If this is fully implemented by Demo 2)
2. User Interaction with Mobile App: User can access and edit Profile Page, view pending and past transactions 
3. Transaction Contract: User initiates contract with Organization using signature 
4. Data Control: User decides what data to show and hide from organization
5. Organization Data: User Data from Organization can be accessed and stored by the user

**Reply by JohanCSmit** [reply]
This sounds perfect.
Let's split #2 into 2 use cases (access and edit profile page  along with  viewing past and pending transactions)

With #3, I think we changed it to the organization initiating the creation of credentials, so we can just reword it a little.

Otherwise this sounds great, we can start moving forward with the diagrams as well.

**Reply by JohanCSmit** [reply]
@RebeccaPillay 

I added Demo 2 :https://github.com/COS301-SE-2022/Exploring-Self-Sovereign-Identity/wiki/Demo-2
to our wiki, you can add the use cases there beneath the older ones.

There are also more specific/updated FR that you can view and use, maybe provide some feedback too please :)

**Reply by RebeccaPillay** [reply]
Okay Great!

So adjusted:

1. User Sign-up: User signs up on SSI mobile app (with Biometrics - If this is fully implemented by Demo 2)
2. User Interaction with Mobile App: User can access and edit Profile Page
3. User can view pending and past transactions
4. Transaction Contract: Organization initiates creation of credentials when User signs contract 
5. Data Control: User decides what data to show and hide from organization
6. Organization Data: User Data from Organization can be accessed and stored by the user

Will take a look at the new FRs and provide feedback shortly.

**Reply by RebeccaPillay** [reply]
The new FRs are perfect. Aligns well with the spec and format.
